qa/mc_sc_connected_nodes.py

```diff
@@ -80,63 +80,69 @@ def run_test(self):
-                                       [genesis_account[1]], [wallet_balance], False)
-        check_mainchan_block_inclusion(second_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block,
-                                       [genesis_account[1]], [wallet_balance], False)
-
-        try:
-            check_mainchan_block_inclusion(third_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block,
-                                           [genesis_account[1]], [wallet_balance], False)
-            # SC node 2 should not to know information about blocks generated from MC node 0
-            # Otherwise the test must to fail
-            assert_equal(1, 2)
-        except AssertionError:
-            print "SC node 2 doesn't include block generated from MC node 0."
-            connect_nodes_bi(self.nodes, 0, 1)
-            self.sync_all()
-            connect_sc_nodes(self.sc_nodes[0], 1)
-            connect_sc_nodes(self.sc_nodes[0], 2)
-            self.sc_sync_all()
-            check_mainchan_block_inclusion(third_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block,
-                                            [genesis_account[1]], [wallet_balance], False)
-            block_hash = second_mainchain_node.generate(1)
-            self.sync_all()
-            second_mainchain_node_new_block = second_mainchain_node.getblock(block_hash[0])
-            third_sidechain_node.block_generate(json.dumps({"number":1}))
-            self.sc_sync_all()
-            check_mainchan_block_inclusion(first_sidechain_node, sidechain_id, 3, 0, second_mainchain_node_new_block,
-                                           [genesis_account[1]], [wallet_balance], False)
-            check_mainchan_block_inclusion(second_sidechain_node, sidechain_id, 3, 0, second_mainchain_node_new_block,
-                                           [genesis_account[1]], [wallet_balance], False)
-            check_mainchan_block_inclusion(third_sidechain_node, sidechain_id, 3, 0, second_mainchain_node_new_block,
-                                           [genesis_account[1]], [wallet_balance], False)
 
+        # SC node 1 and 2 forge 1 SC block
+        sc_generate_blocks(first_sidechain_node)
+        sc_generate_blocks(second_sidechain_node)
+
+        # verify the block is included inside SC nodes 1 and 2
+        sc_1_mc_block_inclusion = is_mainchain_block_included(first_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block)
+        sc_2_mc_block_inclusion = is_mainchain_block_included(second_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block)
+        assert_true(sc_1_mc_block_inclusion, "The mainchain block is not included for SC node 1.")
+        assert_true(sc_2_mc_block_inclusion, "The mainchain block is not included for SC node 2.")
+
+        # verify the mc block is NOT included inside SC node 3
```

Comment:
`third_sidechain_node` did't generate a SC block, so why we need to check anything?

Reply:
To verify there is no propagation of the blocks between sc node 1 and/or 2 and 3